### ISSUE TITLE
 error: ‘get’ is not a member of ‘pangolin’

### ISSUE THREAD

**Issue by wamiq-reyaz** [stevenlovegrove/Pangolin#725]
I was trying to get DeepSDF setup and that has pangolin as a dependency. I have some very weird errors that crop up when trying to run [this file](https://github.com/facebookresearch/DeepSDF/blob/main/src/PreprocessMesh.cpp).

I changed one thing in `gplatfrom.h` on line 82 where instead of gl.h I import glew.h

> /home/parawr/Projects/scenegraphs/external/DeepSDF/src/PreprocessMesh.cpp:37:21: error: ‘get’ is not a member of ‘pangolin’
           pangolin::get<pangolin::Image<uint32_t>>(it_vert_indices->second);
                     ^~~
/home/parawr/Projects/scenegraphs/external/DeepSDF/src/PreprocessMesh.cpp:37:21: note: suggested alternative: ‘Widget’
           pangolin::get<pangolin::Image<uint32_t>>(it_vert_indices->second);
                     ^~~
                     Widget
/home/parawr/Projects/scenegraphs/external/DeepSDF/src/PreprocessMesh.cpp:37:49: error: expected primary-expression before ‘>’ token
           pangolin::get<pangolin::Image<uint32_t>>(it_vert_indices->second);

Any idea what could be causing this?

**Comment by stevenlovegrove** [comment]
Was it working before your change? Pangolin will already include glew.h so I'm not sure why your change was needed?

pangolin::get was just an alias for C++14 get, or a compat library if that wasn't available. That isn't used any more on the master branch I don't think. If you changed the include order, it probably just wasn't including the compatibility header.

**Comment by xiaoyang-RL9** [comment]
DeepSDF is published in 2019, so I install the version around 2019——v0.6, it works! Hope to help you.

**Comment by wetoo-cando** [comment]
I replaced `pangolin::get` calls with `std::get` calls based on @stevenlovegrove's comment above.
I was then able to build DeepSDF successfully with the Pangolin master branch.

Additionally, I also had to change `prog.SetUniform("ttt", 1.0, 0, 0, 1);` calls to `prog.SetUniform("ttt", 1.0, 0.0, 0.0, 1.0);` (the last three arguments has to be changed to floats) to satisfy the new GLSL API.

Other changes I need to make were to the CMakeLists.txt file:

- Added `set(CMAKE_CXX_STANDARD 17)` <-- same as the one in Pangolin master CMakeLists.txt
- Had to extend the list of `target_link_libraries` to `target_link_libraries(PreprocessMesh PRIVATE CLI11::CLI11 pango_core pango_display pango_geometry pango_glgeometry GLEW cnpy Eigen3::Eigen nanoflann::nanoflann)
` based on the hint from [#81](https://github.com/facebookresearch/DeepSDF/issues/81)
